chain/events/filter/index.go

```diff
@@ -646,16 +849,61 @@ func (ei *EventIndex) CollectEvents(ctx context.Context, te *TipSetEvents, rever
 				if rowsAffected != 1 {
 					log.Warnf("restored %d events but expected only one to exist", rowsAffected)
 				}
+
+				res, err = tx.Stmt(ei.stmtRestoreEventSeen).Exec(
```

Comment:
this block seems misplaced - restoring for the tipset inside the internal loop? shouldn't this be with the `stmtInsertEventsSeen` below? you could even do the insert only if `rowsAffected == 0` after this query

Reply:
Done.